R/generate_cpp.R

```diff
@@ -609,3 +609,34 @@ read_types_csv <- function(path, validation_function, group_by_col) {
 
   group_list_of_lists(parsed_types, group_by_col)
 }
+
+
+#' Get mapping of R names to C++ names for input data
+#'
+#' @param input_csv Path to input csv which is source of truth for these names
+#'
+#' @return A list with r_name, cpp_name, and C++ struct/namespace this
+#'   belongs to
+#' @export
+frogger_input_name_mapping <- function(
```

Comment:
The noisiest thing here is that I have added this function. I was getting a bit annoyed with having multiple mappings of R data to C++ data from the scripts. I think I had just avoided writing that previously as I was being lazy. We should now use this anywhere we need to map between the two names. I'm using this in 2 places atm
1. In script to create the python and C++ standalone test data to map from saved R names into C++ names on disk
2. In script to run the R interface to the model (which is used for testing Python & R produce the same result) to map from the C++ names on disk, back into R names